### PR TITLE
Breadth First Search In Ruby

### LINKED ISSUE #2171: Breadth First Search
- [x] C
- [x] C++
- [x] C# - @prashansag62
- [ ] CoffeeScript
- [x] Dart - @Rukmini-Meda
- [ ] Go - @Aju100
- [x] Java
- [x] JS
- [ ] Kotlin
- [ ] Php - @srimani-programmer
- [x] Python
- [x] Readme
- [x] Ruby - @HardevKhandhar
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by HardevKhandhar**
Fixes #2171 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Breadth First Search in Ruby
- 

#### Languages Used:
- Ruby
- 

#### Files Added:
- Breadth_First_Search.rb
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Breadth_First_Search/Breadth_First_Search.rb [hunk 1]
```diff
@@ -0,0 +1,645 @@
+=begin
+    Code contributed by Hardev Khandhar for GirlScript Summer of Code 2020
+    link: https://github.com/HardevKhandhar
+=end
+
+## Breadth First Search algorithm in Ruby.
+
+def breadth_first_search(adj_matrix, source, destination)
+  node_array = [source]
+  path = []
```
Comment (MastersAbh) [inline Breadth_First_Search/Breadth_First_Search.rb]: Please give 4 space indentation across the file

File: Breadth_First_Search/Breadth_First_Search.rb [hunk 2]
```diff
@@ -0,0 +1,645 @@
...
+
+After appending CHILD ARRAY, NODE ARRAY is: [7, 8, 9, 10, 11, 12, 13, 14, 15, 17]
+The next node to be checked is 17.
+Destination Node 17 Found!
+
+The path between Source Node 0 and Destination Node 17 is:
+ [0, 3, 2, 1, 6, 5, 4, 17].
+
+=end
+
```
Comment (MastersAbh) [inline Breadth_First_Search/Breadth_First_Search.rb]: Give a blank line at the end of the file

**Comment by HardevKhandhar** [comment]
Dear Sir @MastersAbh 
The changes have been made & updated in the file as per the guidelines.
Thank you.

**Comment by MastersAbh** [comment]
Also, please make sure you have no spaces in the last line of the file. The last line should be a blank line.

**Comment by HardevKhandhar** [comment]
Dear Sir @MastersAbh 
There has been a mistake in Commits, So many files have changed.
I am sorry, I am closing this PR and raising a new PR. The changes have been made.
Thank you.

**Pull request closed without merging**